server/services/meilisearch/connector.js

```diff
@@ -123,34 +129,41 @@ module.exports = ({ strapi, adapter, config }) => {
 
       if (!Array.isArray(entries)) entries = [entries]
 
-      const indexUid = config.getIndexNameOfContentType({ contentType })
-      await entries.forEach(async entry => {
-        const sanitized = await sanitizeEntries({
-          entries: [entry],
-          contentType,
-          config,
-          adapter,
-        })
-
-        if (sanitized.length === 0) {
-          const task = await client.index(indexUid).deleteDocument(
-            adapter.addCollectionNamePrefixToId({
-              contentType,
-              entryId: entry.id,
+      const indexUids = config.getIndexNamesOfContentType({ contentType })
+      await Promise.all(
```

Comment:
Should we use `allSettled` here?

Reply:
And in the other places where we make calls for all the indexes...

Reply:
It depends on how we want to handle the rejection of the promises generated by 
contentTypeConfig.transformEntry. 

Use Promise.all if: You want the entire operation to fail as soon as one of the transformEntry calls fails. This is useful when it's critical that all entries must be successfully transformed; if any single transformation fails, the whole batch should be considered failed, and you might want to handle that error specifically, perhaps by logging or retrying.

Use Promise.allSettled if: You want to wait for all promises to settle, regardless of whether they are fulfilled or rejected. This approach is useful when you want to attempt to transform all entries and then deal with them individually, depending on whether each promise was fulfilled or rejected. It allows you to handle each entry's transformation success or failure case separately.

which one do you think we should use ?

Reply:
If we had a way to retry the failed entries would be cool I guess. But we can do it later :)